libraries/config/mechanical_material_data.hpp

```diff
@@ -35,26 +35,78 @@ struct mechanical_material_data<T, 1> final {
 };
 
 template<std::floating_point T>
-struct mechanical_material_data<T, 2> final {
+class mechanical_material_data<T, 2> final {
+    void read_elasticity_parameters(std::array<T, 2>& parameter, const nlohmann::json& config, std::string name, std::array<size_t, 2>& null) {
```

Comment:
```suggestion
    std::array<T, 2> read_elasticity_parameters(const nlohmann::json& config, std::string name) {
```

1. Здесь ситуация не требующая супер производительности, поэтому заполненный параметр можно просто вернуть (скорее всего компилятор это оптимизирует при помощи RVO)
2.  Для проверки null я бы написал отдельную функцию, которая будет чекать перед тем как парсить что-либо
3. Просто из своего опыта могу посоветовать структурировать сигнатуру функций по следующему порядку:
- output параметры, которые передаются по ссылке
- input/output параметры, которые передаются по ссылке (параметры, которые могут влиять на алгоритм, но в то же время меняются после его работы)
- input параметры, передающиеся по константной ссылке
- input параметры, передающиеся по значению
- input параметры, у которых есть дефолтные значения

Я это добавлю в заметку по styleguide